internal/provider/resource_device.go

```diff
@@ -82,6 +82,18 @@ func resourceDevice() *schema.Resource {
 							Type:        schema.TypeString,
 							Optional:    true,
 						},
+						"op_mode": {
+							Description:  "Operating mode of the port, valid values are `switch`, `mirror`, and `aggregate`.",
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"switch", "mirror", "aggregate"}, false),
+							// Default:      "switch",
+						},
+						"aggregate_num_ports": {
```

Comment:
Should we add validation to make sure this is only set if `op_mode` is set to `aggregate`?